### PR TITLE
Enable publishing docker images for releases and tags

### PR DESCRIPTION
**Pull request by aterga**

### CODE REVIEW

**Review by mraszyk — COMMENTED** [review]
Why is this needed?

File: .github/workflows/sns_lifecycle.yml [hunk 1]
```diff
@@ -58,7 +58,7 @@ jobs:
         docker kill $SNS_TESTING_INSTANCE
 
   publish:
-    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    if: ${{ github.event_name != 'pull_request' }}
```
Comment (daniel-wong-dfinity-org) [inline .github/workflows/sns_lifecycle.yml:61]: Wouldn't it be simpler to simply remove the subexpression that uses "main"?

**Review by daniel-wong-dfinity-org — APPROVED** [review]
I do not know how these Github YAML files work, but it seems like even if this change is not correct, it can't be that harmful -> approve.

**Comment by daniel-wong-dfinity-org** [comment]
Also, can you change the title of this to say something like "Build and publish new Docker container image in more cases"? And maybe be more specific about what the new cases are?

**Comment by aterga** [comment]
> Why is this needed?

If this works, it should automatically publish the docker image under a image tag corresponding to a GitHub tag. See the table here https://github.com/docker/metadata-action/blob/master/README.md#basic

File: .github/workflows/sns_lifecycle.yml (hunk 1 above)
Comment (daniel-wong-dfinity-org) [reply .github/workflows/sns_lifecycle.yml:61]: I.e. keep 'push', but just do not restrict based on ref.

Rationale: `!= 'whatever'` encompasses a many things that are not explicitly mentioned.

**Comment by aterga** [comment]
Merging as-is to be able to test this.

**Pull request merged**